session_manager.py

```diff
@@ -1,44 +1,86 @@
 import iitkgp_erp_login.erp as erp
 import requests
 from utils import ERPSession
-
+from typing import Dict
+from datetime import datetime, timedelta
+from threading import Timer
 headers = {
    'timeout': '20',
    'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/51.0.2704.79 Chrome/51.0.2704.79 Safari/537.36',
 }
-session = requests.Session()
+
+
+
+class Session:
+    def __init__(self, session_token: str ="", sso_token: str ="", erp_session: ERPSession = None, login_details: Dict[str, str] = None):
+        self.session_token = session_token
+        self.sso_token = sso_token
+        self.requests_session = requests.Session()
+        self.erp_session = erp_session
+        self.login_details = login_details
+        self.timestamp = datetime.now() 
 
 
 class SessionManager:
-    rollTologinDetailsMap = {}
-    rollToSessionTokenMap = {}
+    sessions: Dict[str, Session] = {}
 
     @staticmethod
     def get_secret_question(roll):
-        sessionToken = erp.get_sessiontoken(session)
-        SessionManager.rollToSessionTokenMap[roll] = sessionToken
-        secret_question = erp.get_secret_question(headers, session, roll)
+        session = SessionManager.sessions.get(roll)
+        if not session:
+            session = Session()
+            SessionManager.sessions[roll] = session
```

Comment:
Storing user sessions with their roll numbers as keys seems unsafe to me. What if I know someone is logged in, and I use their roll number. Maybe create a `session_id` for each session, and relay the session id for the particular user for each request ?

thoughts @proffapt ??

Reply:
Yes the issue is real, we can implement any solution. What you said can also work